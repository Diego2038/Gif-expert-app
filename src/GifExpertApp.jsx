import { useState } from "react";
import { AddCategory } from "./components/AddCategory";

export const GifExpertApp = () => {

  const [categories, setCategories] = useState(['One punch', 'DBZ']);

  const onNewCategory = ( value ) => {
    // console.log( value );
    setCategories( categories => [ value, ...categories ] );
  }

  /* const onAddCategory = () => {
    // setCategories( category => category.push('pepe'));
    console.log('hola')
    setCategories( ( cat ) => ['pepe', ...cat]);
  } */
  
  return (
    <>
    {/* TÃ­tulo */}
      <h1>GifExpertApp</h1>
    
    {/* Input */}
    <AddCategory 
    hola='mundo' 
    // setCategories={ setCategories } 
    onNewCategory={onNewCategory}
    /> 

    {/* Listado de Gifs */}
      {/* Gif item */}
    <ol> 
      { 
      categories.map( category => {
          return (<li key={ category }> {category} </li>)
        }) 
      }
      {/* <li>abc</li>  */}
    </ol>

    </>
  )
}
