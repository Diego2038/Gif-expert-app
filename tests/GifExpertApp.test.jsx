import { fireEvent, render, screen } from "@testing-library/react";

import { GifExpertApp } from "../src/GifExpertApp";
 
describe('test component <GifExpertApp/>', () => {

  test('should to match with the snapshot', () => {
    const { container } = render( <GifExpertApp />);
    screen.debug();
    expect( container ).toMatchSnapshot(); 
  });

  test('should to return more components when send an input event', () => {
    render( <GifExpertApp/> );
    const onNewCategory = jest.fn( () => {

    });
    const value = 'Dragon Ball';

    const input = screen.getByRole('textbox');
    const form = screen.getByRole('form');
    fireEvent.input( input, { target: { value }});
    fireEvent.submit( form );

    // screen.debug();
    const titles = screen.getAllByRole('heading', { level: 3});
    expect( titles.length).toBe(2);
  });

  test('don\'t should add a new category if a previous category exists', () => {
    render( <GifExpertApp /> );
    const input = screen.getByRole('textbox');
    const form = screen.getByRole('form');
    const value = 'One punch';

    fireEvent.input( input, { target: { value }});
    fireEvent.submit( form );

    expect( screen.getAllByRole('heading', {level: 3}).length ).toBe(1);
  })






});