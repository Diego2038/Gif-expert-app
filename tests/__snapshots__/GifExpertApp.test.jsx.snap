// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component <GifExpertApp/> should to match with the snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="foorm"
  >
    <input
      placeholder="Search a gif"
      type="text"
      value=""
    />
    <h6>
      mundo
    </h6>
  </form>
  <ol>
    <h3>
      One punch
    </h3>
    <h2>
      Cargando...
    </h2>
    <div
      class="card-grid"
    />
  </ol>
</div>
`;
